lib/rpm_package.py

```diff
@@ -72,26 +72,31 @@ def write_content(self):
             file_.write(self._content)
         self._cached_tags = dict()
 
-    def query_tag(self, tag_name):
+    def query_tag(self, tag_name, extra_args=""):
         """
         Queries the spec file for a tag's value.
         Cached content not yet written to the file is not considered.
 
         Args:
             tag_name (str): name of the tag to query
+            extra_args (str): extra arguments to append to the command
 
         Returns:
             str: tag value
         """
-        if tag_name not in self._cached_tags:
-            tag_value = utils.run_command(
-                "rpmspec --srpm -q --qf '%%{%s}' %s 2>/dev/null" % (
-                    tag_name.upper(), self.path)).strip()
+        if tag_name in self._cached_tags:
+            tag_value = self._cached_tags[tag_name]
+        else:
+            command = "rpmspec --srpm -q --qf '%%{%s}' %s %s 2>/dev/null" % (
+                    tag_name.upper(), self.path, extra_args)
+            tag_value = utils.run_command(command).strip()
             if tag_value == "(none)":
                 tag_value = None
-            self._cached_tags[tag_name] = tag_value
+            if not extra_args:
+                # Extra arguments may define macros that alter the tags
+                self._cached_tags[tag_name] = tag_value
```

Comment:
This way, if we call `query_tag` with `extra_args` and later without it, we might have an unexpected result.
This could happen if we wanted to overwrite a value that is present in the spec file, for example.